californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -679,17 +679,20 @@ private void reject(final RawData raw, final MessageFormatException cause) {
 			rst.setToken(new byte[0]);
 			rst.setDestination(raw.getAddress());
 			rst.setDestinationPort(raw.getPort());
-			for (MessageInterceptor interceptor : interceptors) {
-				interceptor.sendEmptyMessage(rst);
-			}
-			connector.send(serializer.serializeEmptyMessage(rst));
+
+			reject(rst);
```

Comment:
> I don't think doing this in the Matcher will work. 

> From my understanding:
> though the PING/PONG depends on the protocol flavor (TCP/UDP), we should try to move it to the Matcher and Stack and handle it there flavor aware.

I ment this very general: if there is something depending on the "flavor" (currently tcp/udp), then we should try to move it in the already flavor aware components (Matcher and Stack). Which one (or both), is then left to the implementor. 
I have also no problems with "if (TCP/UDP)" somewhere else to fix a bug, if we then but a reminder issue to solve it later without "if".

> One option that I like a little more now, is to just send RST through the stack. I think that should have no side effect, because non of the layers do anything stateful with RST message (and in fact, may have positive side effect by clearing any state in the matcher).
> 
> And that will solve the problem of infinite ping pong, because it TcpStack will correctly ignore RST message. Thoughts?
> 

I like this idea.